spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,161 @@
+require_relative 'spec_helper.rb'
+require 'date'
+
+describe "Reservation Manager" do
+  before do
+    @frontdesk = Hotel::ReservationManager.new
+  end
+
+  it "should create an instance of the reservation Manager" do
+    expect(@frontdesk).must_be_instance_of Hotel::ReservationManager
+  end 
+
+
+  it "displays all available rooms" do
+    expect(@frontdesk.hotel_rooms).must_be_instance_of(Array)
+    expect(@frontdesk.hotel_rooms.first).must_equal(1)
+    expect(@frontdesk.hotel_rooms[3]).must_equal(4)
+    expect(@frontdesk.hotel_rooms.length).must_equal(20)
+  end
+
+  describe "book reservation method" do
+    before do
+      @frontdesk = Hotel::ReservationManager
+      @frontdesk.book_reservation(1, Date.new(2019,3,15), Date.new(2019,3,18))
+      @frontdesk.book_reservation(1, Date.new(2019,4,10), Date.new(2019,3,12))
+      @frontdesk.book_reservation(2, Date.new(2019,3,12), Date.new(2019,3,17))
+    end
+
+  end
+
+  describe 'room availability method' do
+    it "returns available rooms" do 
+      [*1..5].each do |room|
+        @frontdesk.book_reservation(room, Hotel::DateRange.new(Date.new(2019,3,15),Date.new(2019,3,18)))
+      end
+      expect(@frontdesk.room_availability(Date.new(2019,3,15), Date.new(2019,3,18))).must_equal([*6..20])
+    end
+
+    it "returns empty array if all rooms are booked" do
+      [*1..20].each do |room|
+        @frontdesk.book_reservation(room, Hotel::DateRange.new(Date.new(2019,3,15),Date.new(2019,3,18)))
+      end
+      expect(@frontdesk.room_availability(Date.new(2019,3,15), Date.new(2019,3,18))).must_equal([])
+      expect(@frontdesk.room_availability(Date.new(2019,3,11), Date.new(2019,3,16))).must_equal([])
+    end
+  end
+
+  describe "book reservation method" do
+    before do
+      @reservations = []
+      @test_range = Hotel::DateRange.new(Date.new(2019,3,11), Date.new(2019,3,15))
+    end
+     
+    it "stores booked reservations in reservations array" do
+      res1 = @frontdesk.book_reservation(1, @test_range)
+      @reservations << res1
+
+      expect(@reservations.length).must_equal(1)    
+    end
+
+    it "raises an error if the room is not available" do
```

Comment:
Your room here is still available so the `ArgumentError` is not being raised.  This is a good start on testing the double-book issue.